### ISSUE TITLE
Segfault when processing one client message.

### ISSUE THREAD

**Issue by Bertram25** [OpenDungeons/OpenDungeons#69]
I got this segfault on turn 1251. Nothing obvious related.

[bt]: (1) ./OpenDungeons.bin : GameMap::getAnimatedObject(std::string const&)+0x5c [0x582ddc]
[bt]: (2) ./OpenDungeons.bin : GameMap::getAnimatedObject(std::string const&)+0x5c [0x582ddc]
[bt]: (3) ./OpenDungeons.bin : ODClient::processOneClientSocketMessage()+0x4a4e [0x5c52ae]
[bt]: (4) ./OpenDungeons.bin : ODClient::processClientSocketMessages()+0x20 [0x5cf860]
[bt]: (5) ./OpenDungeons.bin : ODFrameListener::frameStarted(Ogre::FrameEvent const&)+0x192 [0x5d4ba2]
[bt]: (6) /usr/lib/x86_64-linux-gnu/libOgreMain.so.1.9.0 : Ogre::Root::_fireFrameStarted(Ogre::FrameEvent&)+0x54 [0x7ffd9eb99694]
[bt]: (7) /usr/lib/x86_64-linux-gnu/libOgreMain.so.1.9.0 : Ogre::Root::_fireFrameStarted()+0x21 [0x7ffd9eb99be1]
[bt]: (8) /usr/lib/x86_64-linux-gnu/libOgreMain.so.1.9.0 : Ogre::Root::renderOneFrame()+0x9 [0x7ffd9eb99cd9]
[bt]: (9) /usr/lib/x86_64-linux-gnu/libOgreMain.so.1.9.0 : Ogre::Root::startRendering()+0x35 [0x7ffd9eb99d35]
[bt]: (10) ./OpenDungeons.bin : ODApplication::ODApplication()+0x3fc [0x5bfe3c]
[bt]: (11) ./OpenDungeons.bin : main+0x1e [0x466abe]
[bt]: (12) /lib/x86_64-linux-gnu/libc.so.6 : __libc_start_main+0xf5 [0x7ffd9b8f9b45]
[bt]: (13) ./OpenDungeons.bin() [0x467e84]

**Comment by hwoarangmy** [comment]
getAnimatedObject is not recursive. How can you have 2 calls ?

**Comment by Bertram25** [comment]
I wish I knew. :S 

Might there be one call for the server, one for the client? I might be completely wrong here, though.

**Comment by hwoarangmy** [comment]
No, if the call comes from ODClient, you are on client side. Until now, there is no hack and the server and client gamemaps are really separated. Every communication goes through ODPackets.
This callstack could be for changing destination or animation state. The kind of call that happens more than 10 times per turn :)

**Comment by Bertram25** [comment]
This might just be about adding a few check on pointer received by this call and report info in the log as a first step. :)

Just to point it out while I see it, the very way the getAnimatedObject(int index) is done is bad, btw:

```
MovableGameEntity* GameMap::getAnimatedObject(int index)
{
    MovableGameEntity* tempAnimatedObject = animatedObjects[index];
    return tempAnimatedObject;
}
```

It should be redone the way its const std::string& counter part is done and return NULL on problems with the parameter instead, or be removed. and replaced with a std::vector<>& getter instead maybe as it is the only way it is used?

Also, the bt I have given is based on @tomluchowski's code and might give different result than a true gdb session. I'll play using gdb to see if something different is shown.

**Comment by Bertram25** [comment]
Ah, and here:
https://github.com/OpenDungeons/OpenDungeons/blob/development/source/GameMap.cpp#L390

`MovableGameEntity* mge = animatedObjects[i];`
`mge` isn't check against being NULL.

**Comment by hwoarangmy** [comment]
Yes, it has never been. I've splitted this in 2 to be able to have a look at the entity (I can't read vectors with MinGW gdb). But before, it was animatedObjects[i]->getName().compare(name) so it is the same. But testing it would be to accept NULL values to be added. As you said, we should add some logs if we see that this bug occurs often (and as I've seen, it is at least the 2nd time it is posted) and see where this NULL value comes from.

**Comment by hwoarangmy** [comment]
This issue is probably the same as the one referenced by akien is #47

**Comment by Bertram25** [comment]
Yeah, probably. As the #47 is messy, once your latest pull request is in, are there any open issues left in it, this one apart? (If so, I think we'll then be able to close it.)

**Comment by hwoarangmy** [comment]
I could not reproduce the crashes with the long nicknames so I'm not sure. But we can close it and re-open when we will try the next lan if the problem is still there.

**Comment by Bertram25** [comment]
Agreed.

**Comment by Bertram25** [comment]
@hwoarangmy 
I believe it fixed with: https://github.com/OpenDungeons/OpenDungeons/commit/301dcab35c767466677a5bb6761e1326e72ab53a

If @akien-mageia is ok, we'll close this one as well after some tests. Thanks a lot for the hard work done here. It's marvelous to think we can play an open keeper game live!

**Comment by hwoarangmy** [comment]
> I believe it fixed with: 301dcab

I think so

**Comment by hwoarangmy** [comment]
I'm closing it as it is supposed to be fixed.
